src/Home/Home.spec.tsx

```diff
@@ -355,16 +386,76 @@ describe("Home", () => {
           const alert = jest.spyOn(Alert, "alert")
 
           fireEvent.press(fixProximityTracingButton)
-          await waitFor(() => {
-            expect(alert).toHaveBeenCalledWith(
-              "Authorize in Settings",
-              "To activate Proximity Tracing, authorize COVID-19 Exposure Logging in the Settings app",
-              [
-                expect.objectContaining({ text: "Back" }),
-                expect.objectContaining({ text: "Open Settings" }),
-              ],
-            )
-          })
+          expect(alert).toHaveBeenCalledWith(
+            "Authorize in Settings",
+            "To activate Proximity Tracing, authorize COVID-19 Exposure Logging in the Settings app",
+            [
+              expect.objectContaining({ text: "Back" }),
+              expect.objectContaining({ text: "Open Settings" }),
+            ],
+          )
+        })
+      })
+    })
+    describe("when the device requires location permissions", () => {
+      it("adds the location status to the home screen", () => {
+        const locationIsNeeded = true
+        ;(useLocationIsNeededStatus as jest.Mock).mockReturnValueOnce(
+          locationIsNeeded,
+        )
+
+        const { getByTestId } = render(<Home />)
+
+        const locationStatusContainer = getByTestId(
+          "home-location-status-container",
+        )
+
+        const locationText = within(locationStatusContainer).getByText(
+          "Location",
+        )
+
+        expect(locationText).toBeDefined()
+      })
+      describe("when the location permissions are on", () => {
+        it("the location permissions are shown as enabled", () => {
+          const locationIsNeeded = true
+          ;(useLocationIsNeededStatus as jest.Mock).mockReturnValueOnce(
+            locationIsNeeded,
+          )
+          const isLocationOn = true
+          ;(useLocationStatus as jest.Mock).mockReturnValueOnce(isLocationOn)
+
+          const { getByTestId } = render(<Home />)
+
+          const locationStatusContainer = getByTestId(
+            "home-location-status-container",
+          )
+          const enabledText = within(locationStatusContainer).getByText(
+            "Enabled",
+          )
+
+          expect(enabledText).toBeDefined()
+        })
+      })
+      describe("when the location permissions are off", () => {
```

Comment:
```suggestion
      describe("and the location permissions are off", () => {
        it("shows the locations permissions as disabled", () => {
```

And a thought: if we have the second two tests here, the permissions on and off cases, do we need the first test asserting that the location is present? or do the second two cases sufficiently cover this?

Reply:
I actually don't know. I'd be curious what your recommendation is here because I'm very much learning about how to properly test.

Reply:
mostly brought it up to get you thinking about it! as this is a topic that comes up a lot in testing. that is, what is the appropriate amount of coverage for a given feature? and how do we best achieve that coverage?

 On the one hand we want maximum coverage as to feel confident that our system works, but on the other hand if we over cover, then our test suite becomes slow and is a drag on productivity. So its a tradeoff we need to make and have some opinions on what are best practices around what to test and why.

typically we want to cover as much of what is possible as possible, but no more than the minimum. and its common for one test to have overlap with others, which is what is happening here as far as i can tell. In this case, imo, it seems like the second two tests fully cover the first, so the first is redundant, and could be removed. Of course I defer to you as the author of this commit to what makes the most sense though.

let me know if youd like to chat about it more irl!

Reply:
This is super helpful, thank you! I agree. Will remove that test.